### PR TITLE
Feature - Add Queue Data Structure Implementation

### PR DESCRIPTION
**Pull request by EricNeves**
📋 Overview
This pull request introduces the implementation of a queue data structure, providing essential operations and functionality.

✅ Features
- Enqueue and Dequeue: Allows adding and removing elements from the queue.
- IsEmpty: Checks if the queue is empty.
- Size: Returns the number of elements in the queue.
- Peek: Views the front element without removing it.
- Clear: Empties all elements from the queue.
- ToString: Returns a string representation of the queue.

🔍 Details
- Implemented core methods: <code>enqueue</code>, <code>dequeue</code>, <code>isEmpty</code>, <code>size</code>, <code>peek</code>, <code>clear</code>, and <code>toString</code>.
-  Implemented tests: Ensured the reliability and correctness of the queue operations.

Please review the changes and let me know if there are any suggestions or improvements needed. Thank you for your time and consideration! 🚀

### CODE REVIEW

File: tests/DataStructures/QueueTest.php
Comment (darwinz) [inline tests/DataStructures/QueueTest.php:1]: Let's add a test that asserts the value of a dequeued element

Comment (EricNeves) [reply tests/DataStructures/QueueTest.php:1]: That's ok, thanks!

Comment (EricNeves) [reply tests/DataStructures/QueueTest.php:1]: I've added the test and committed the changes. Thanks!

File: tests/DataStructures/QueueTest.php [hunk 1]
```diff
@@ -0,0 +1,152 @@
+<?php
+use PHPUnit\Framework\TestCase;
```
Comment (darwinz) [inline tests/DataStructures/QueueTest.php:2]: To follow PSR-12 guidelines, to make the linter happy - should have one blank line between header blocks.

```suggestion
<?php

use PHPUnit\Framework\TestCase;
```

**Review by darwinz — APPROVED** [review]
Great addition, thank you!

**Pull request merged**